extensions/amp-share-tracking/0.1/amp-share-tracking.js

```diff
@@ -15,41 +15,173 @@
  */
 
 import {isExperimentOn} from '../../../src/experiments';
-import {getService} from '../../../src/service';
+import {fromClass} from '../../../src/service';
+import {xhrFor} from '../../../src/xhr';
+import {viewerFor} from '../../../src/viewer';
 import {dev} from '../../../src/log';
 
 /** @private @const {string} */
 const TAG = 'amp-share-tracking';
 
-class AmpShareTracking extends AMP.BaseElement {
+/**
+ * @export
+ * @typedef {{
+ *   outgoingFragment: string
+ * }}
+ */
+let ShareTrackingPostResponseDef;
+
+export class AmpShareTracking extends AMP.BaseElement {
+  /**
+    * @return {boolean}
+    * @private
+    */
+  isExperimentOn_() {
+    return isExperimentOn(this.getWin(), TAG);
+  }
+
+  /** @override */
+  isLayoutSupported(unusedLayout) {
+    return true;
+  }
+
+  /** @override */
+  createdCallback() {
+    if (!this.isExperimentOn_()) {
+      dev.warn(TAG, `TAG ${TAG} disabled`);
+      return;
+    }
+
+    /** @private @const {!Window} */
+    this.win_ = this.getWin();
+
+    /** @private @const {!ShareTrackingService} */
+    this.shareTrackingService_ = getShareTrackingService_(this.win_);
+  }
 
   /** @override */
   buildCallback() {
-    this.isExperimentOn_ = isExperimentOn(this.getWin(), TAG);
-    if (!this.isExperimentOn_) {
+    if (!this.isExperimentOn_()) {
       dev.warn(TAG, `TAG ${TAG} disabled`);
       return;
     }
+
+    this.vendorHref = this.element.getAttribute('data-href');
+    dev.fine(TAG, 'vendorHref: ', this.vendorHref);
+
+    return Promise.all([
+      this.shareTrackingService_.getIncomingFragment(),
+      this.shareTrackingService_.getOutgoingFragment(this.vendorHref)]).then(
+        results => {
+          this.incomingFragment = results[0];
+          this.outgoingFragment = results[1];
+          dev.fine(TAG, 'get incoming fragment: ' + this.incomingFragment);
+          dev.fine(TAG, 'get outgoing fragment: ' + this.outgoingFragment);
+        });
   }
 }
 
 /**
- * ShareTrackingService processes the incoming and outcoming url fragment
+ * ShareTrackingService processes the incoming and outgoing url fragment
  */
 export class ShareTrackingService {
```

Comment:
After talking to @lannka, we agreed that there's no need for a separate class `ShareTrackingService `. I can put  all the methods inside `AmpShareTracking`